cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -222,6 +254,23 @@ public void resume() throws Exception {
   protected void shutDown() throws Exception {
     httpService.stopAndWait();
     deleteLocalDatasets();
+    destroyWorkflow();
```

Comment:
can deleteLocalDatasets() throw exceptions? if so we need to catch and make sure destroyWorkflow is run